pkg/parse/root_test.go

```diff
@@ -1788,3 +1788,145 @@ func TestSummarizeErrors(t *testing.T) {
+
+metadata.name: foo-ns
+group:
+version: v1
+kind: Namespace
+
+For more information, see https://g.co/cloud/acm-errors#knv1060`
+	const kptManagementConflictMessage = `KNV1060: The "other-manager" reconciler cannot manage resources declared in another repository. Remove the declaration for this resource from either the current repository, or the managed repository.
+
+metadata.name: foo-ns
+group:
+version: v1
+kind: Namespace
+
+For more information, see https://g.co/cloud/acm-errors#knv1060`
+	// TODO: CurrentManagerError currently seems to print an incorrect error message
+	const currentManagerMessage = `KNV1060: The "this-manager" reconciler cannot manage resources declared in another repository. Remove the declaration for this resource from either the current repository, or the repository managed by "this-manager".
+
+metadata.name: foo-ns
+group:
+version: v1
+kind: Namespace
+
+For more information, see https://g.co/cloud/acm-errors#knv1060`
+	// TODO: ConflictingManagerError currently seems to print an incorrect error message
+	const conflictingManagerMessage = `KNV1060: The "this-manager" reconciler detects a management conflict for a resource declared in another repository. Remove the declaration for this resource from either the current repository, or the repository managed by "this-manager".
+
+metadata.name: foo-ns
+group:
+version: v1
+kind: Namespace
+
+For more information, see https://g.co/cloud/acm-errors#knv1060`
+	testutil.AssertEqual(t, thisManagerMessage, conflictError.ToCSE().ErrorMessage)
+	testutil.AssertEqual(t, currentManagerMessage, currentManagerError.ToCSE().ErrorMessage)
+	testutil.AssertEqual(t, otherManagerMessage, invertedConflictError.ToCSE().ErrorMessage)
+	testutil.AssertEqual(t, conflictingManagerMessage, conflictingManagerError.ToCSE().ErrorMessage)
+	testutil.AssertEqual(t, kptManagementConflictMessage, kptManagementConflictError.ToCSE().ErrorMessage)
+	testCases := map[string]struct {
+		thisSyncErrors []v1beta1.ConfigSyncError
+		expectedErrors []v1beta1.ConfigSyncError
+	}{
+		"empty errors": {
+			thisSyncErrors: []v1beta1.ConfigSyncError{},
+			expectedErrors: []v1beta1.ConfigSyncError{
+				conflictError.ToCSE(),
+			},
+		},
+		"unchanged conflict error": {
+			thisSyncErrors: []v1beta1.ConfigSyncError{
+				conflictError.ToCSE(),
+			},
+			expectedErrors: []v1beta1.ConfigSyncError{
+				conflictError.ToCSE(),
+			},
+		},
+		"prepend with other error": {
+			thisSyncErrors: []v1beta1.ConfigSyncError{
+				{ErrorMessage: "foo"},
+			},
+			expectedErrors: []v1beta1.ConfigSyncError{
+				conflictError.ToCSE(),
+				{ErrorMessage: "foo"},
+			},
+		},
+		"unchanged CurrentManagerError": {
+			thisSyncErrors: []v1beta1.ConfigSyncError{
+				currentManagerError.ToCSE(),
+			},
+			expectedErrors: []v1beta1.ConfigSyncError{
+				currentManagerError.ToCSE(),
+			},
+		},
+		// TODO: the follow testcases document the current behavior, but probably isn't the desired behavior
```

Comment:
So it doesn't actually prepend them?

Reply:
It does prepend, I assume the intent is to de-dupe - which should be fixed by your change